src/main/scala/higherkindness/rules_scala/common/worker/CancellableTask.scala

```diff
@@ -16,12 +16,12 @@ import scala.util.Try
  * Heavily inspired by the following: https://github.com/NthPortal/cancellable-task/tree/master
  * https://stackoverflow.com/a/39986418/6442597
  */
-class CancellableTask[S] private (fn: => S) {
+class CancellableTask[S] private (fn: Function1[Function0[Boolean], S]) {
```

Comment:
Nit: What's wrong with `(() => Boolean) => S`?

Reply:
I find this more readable. I played around with that syntax for a few minutes and ultimately decided the FunctionX syntax was more readable once things got nested.

Reply:
Fair enough.